weave/trace_server/refs_internal.py

```diff
@@ -17,16 +17,82 @@
 OBJECT_ATTR_EDGE_NAME = "attr"
 TABLE_ROW_ID_EDGE_NAME = "id"
 
+valid_edge_names = (
+    DICT_KEY_EDGE_NAME,
+    LIST_INDEX_EDGE_NAME,
+    OBJECT_ATTR_EDGE_NAME,
+    TABLE_ROW_ID_EDGE_NAME,
+)
+
 
 class InvalidInternalRef(ValueError):
     pass
 
 
+def quote_select(s: str, quote_chars: tuple[str, ...] = ("/", ":", "%")) -> str:
```

Comment:
Are we missing the whitespace char here?  It could be common in e.g. names

Reply:
the objective is not to make a valid URL for browsers, bur rather to ensure that we can parse the ref URI. in which case whitespace is perfectly valid